packages/react-components/src/components/HorizontalGallery.tsx

```diff
@@ -58,15 +62,20 @@ export const HorizontalGallery = (props: HorizontalGalleryProps): JSX.Element =>
   const numberOfChildren = React.Children.count(children);
   const lastPage = Math.ceil(numberOfChildren / childrenPerPage) - 1;
 
+  const indexesArray: number[][] = useMemo(() => {
```

Comment:
Do we need this to be 2d array? I guess we only need the array from the current page one?
So return bucketize([...Array(numberOfChildren).keys()], childrenPerPage)[page]